src/Downloader.php

```diff
@@ -59,12 +59,17 @@ public function __construct(Composer $composer, IoInterface $io, $rfs)
         if (null === $endpoint = $composer->getPackage()->getExtra()['symfony']['endpoint'] ?? null) {
             $this->endpoints = self::DEFAULT_ENDPOINTS;
 
-            if (!filter_var(getenv('FLEX_SERVERLESS'), \FILTER_VALIDATE_BOOLEAN)) {
+            if (!$this->isServerlessEnabled()) {
                 $this->endpoints = null;
                 $this->legacyEndpoint = 'https://flex.symfony.com';
             }
         } elseif (\is_array($endpoint) || false !== strpos($endpoint, '.json') || 'flex://defaults' === $endpoint) {
             $this->endpoints = array_values((array) $endpoint);
+            if (false !== strpos($endpoint, '.json')) {
+                $this->endpoints[] = 'flex://defaults';
+            }
+        } elseif ($this->isServerlessEnabled() && \is_string($endpoint)) {
```

Comment:
I think this case should be removed since when this is a string and it doesn't contain `.json`, the it means a legacy endpoint is configured.
If confirmed, then let's remove the `isServerlessEnabled` method.

Reply:
It does mean that, but this PR is exactly about that.

If we use a string which does not end in `.json` in `extra.symfony.endpoint` like this:

```
"extra": {
    "symfony": {
        "endpoint": "https://api.github.com/...",
        "allow-contrib": true,
        "require": "5.3.*"
    }
}
```

Flex will consider it a legacy endpoint and serverless will not work. My goal is to have the above config work even in serverless mode (with the fallback to `flex://defaults`), which is activated by the env var. If the env var is present, only then should this be considered a legacy endpoint.

I hope I'm making myself clear enough :)

Reply:
We can remove the condition if you do not wish to support any string (even ones without `.json`) as serverless endpoints, however, this will in time be supported implicitly once you remove the legacy mode completely.

Reply:
we're going to remove the FLEX_SERVERLESS env var in a few weeks, so better drop this now

Reply:
Done!